### DISCUSSION TITLE
Is there a way to safely downgrade from 3.0 to 2.94 and keep all the current resume information?

### DISCUSSION THREAD

**Discussion by Endemoniada** [transmission/transmission#2457] [Q&A] [answered]
I'm trying to downgrade to 2.94 on Arch Linux, but when I do the old resume files no longer have the correct download path in them. They've all been set to the default download location, which means they can't resume properly.

Is there a way to safely downgrade from 3.0 to 2.94 and keep all the current resume information?

**Comment by ckerr** [comment, marked as answer]
Probably not? The file naming scheme for .resume and .torrent files changed to fix https://github.com/transmission/transmission/issues/122.

You might be able to go in and rename the files back to the old names by hand, but there's no way inside of the Transmission application to downgrade the files. 2.94 doesn't have any knowledge of the changes that 3.00 made.

**Comment by linesteppr** [comment]
I just ran into this today after upgrading to Raspbian Bullseye and things running horribly.  I'm deciding between working on a script to rename the new resume files to the old style and seeing if that works or just going back to 3.0.

**Comment by linesteppr** [comment]
YES!  I wrote a python script to read the torrent name from the resume file and create a copy of the 3.0 resume file with a 2.x resume file name.

```
# requires "modern-bencode"
# pip install modern-bencode
# make a directory in the resume folder called "new"
# run this script from the directory above the "resume" directory
# the new files may not have the correct ownership so change that as needed after running the script

from os import path
from pathlib import Path
import os
from bencode import decode
import shutil

os.chdir(Path.cwd() / 'resume')
for rfile in Path.cwd().iterdir():
    if Path(rfile).suffix == '.resume':
        with ((open(Path(rfile), "rb"))) as rdata:
            b = rdata.read()
            d = decode(b)
            nfile = Path(os.getcwd()+'/new/'+d[bytes("name",'utf-8')].decode('utf-8')+'.'+rfile.name[:16]+'.resume')
            # grab the torrent name from inside the resume file and concatenate it with the first 16 characters of the resume file name (the hash)
        shutil.copy2(rfile,nfile)

# the warnings come after the spells.
```
